.github/workflows/ci-core.yml

```diff
@@ -200,7 +209,8 @@ jobs:
 
   scan:
     name: SonarQube Scan
-    needs: [core-complete]
+    needs: [core]
+    if: ${{ always() }}
```

Comment:
Could if/always be simplified away? Is that different then omitting this line entirely?

Reply:
Might need @HenryNguyen5 to chime in and confirm but IIRC `core-complete` only runs if `core` is successful. By making SonarQube dependent on `core` by default it will only run if `core` is successful. But the if/always [should make sure](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#example-not-requiring-successful-dependent-jobs) that we run SonarQube even if `core` fails.